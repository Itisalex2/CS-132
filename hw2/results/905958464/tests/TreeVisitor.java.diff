0a1
> Program type checked successfully
