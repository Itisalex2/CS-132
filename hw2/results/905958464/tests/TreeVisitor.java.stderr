Visiting main class. Added main class: ClassInfo{name='TreeVisitor', fields={}, methods={main: MethodInfo{name='main', returnType='void', parameters={a: STRING_ARRAY}, localVariables={}}}}
Visiting class: ClassInfo{name='TV', fields={}, methods={}}
Added method: Start of type INT to class TV
Added local variable: root of type ClassType(Tree) to method Start
Added local variable: ntb of type BOOLEAN to method Start
Added local variable: nti of type INT to method Start
Added local variable: v of type ClassType(MyVisitor) to method Start
Visiting class: ClassInfo{name='Tree', fields={}, methods={}}
Added field: left of type ClassType(Tree) to class Tree
Added field: right of type ClassType(Tree) to class Tree
Added field: key of type INT to class Tree
Added field: has_left of type BOOLEAN to class Tree
Added field: has_right of type BOOLEAN to class Tree
Added field: my_null of type ClassType(Tree) to class Tree
Added method: Init of type BOOLEAN to class Tree
Added method: SetRight of type BOOLEAN to class Tree
Added method: SetLeft of type BOOLEAN to class Tree
Added method: GetRight of type ClassType(Tree) to class Tree
Added method: GetLeft of type ClassType(Tree) to class Tree
Added method: GetKey of type INT to class Tree
Added method: SetKey of type BOOLEAN to class Tree
Added method: GetHas_Right of type BOOLEAN to class Tree
Added method: GetHas_Left of type BOOLEAN to class Tree
Added method: SetHas_Left of type BOOLEAN to class Tree
Added method: SetHas_Right of type BOOLEAN to class Tree
Added method: Compare of type BOOLEAN to class Tree
Added local variable: ntb of type BOOLEAN to method Compare
Added local variable: nti of type INT to method Compare
Added method: Insert of type BOOLEAN to class Tree
Added local variable: new_node of type ClassType(Tree) to method Insert
Added local variable: ntb of type BOOLEAN to method Insert
Added local variable: current_node of type ClassType(Tree) to method Insert
Added local variable: cont of type BOOLEAN to method Insert
Added local variable: key_aux of type INT to method Insert
Added method: Delete of type BOOLEAN to class Tree
Added local variable: current_node of type ClassType(Tree) to method Delete
Added local variable: parent_node of type ClassType(Tree) to method Delete
Added local variable: cont of type BOOLEAN to method Delete
Added local variable: found of type BOOLEAN to method Delete
Added local variable: ntb of type BOOLEAN to method Delete
Added local variable: is_root of type BOOLEAN to method Delete
Added local variable: key_aux of type INT to method Delete
Added method: Remove of type BOOLEAN to class Tree
Added local variable: ntb of type BOOLEAN to method Remove
Added local variable: auxkey1 of type INT to method Remove
Added local variable: auxkey2 of type INT to method Remove
Added method: RemoveRight of type BOOLEAN to class Tree
Added local variable: ntb of type BOOLEAN to method RemoveRight
Added method: RemoveLeft of type BOOLEAN to class Tree
Added local variable: ntb of type BOOLEAN to method RemoveLeft
Added method: Search of type INT to class Tree
Added local variable: current_node of type ClassType(Tree) to method Search
Added local variable: ifound of type INT to method Search
Added local variable: cont of type BOOLEAN to method Search
Added local variable: key_aux of type INT to method Search
Added method: Print of type BOOLEAN to class Tree
Added local variable: ntb of type BOOLEAN to method Print
Added local variable: current_node of type ClassType(Tree) to method Print
Added method: RecPrint of type BOOLEAN to class Tree
Added local variable: ntb of type BOOLEAN to method RecPrint
Added method: accept of type INT to class Tree
Added local variable: nti of type INT to method accept
Visiting class: ClassInfo{name='Visitor', fields={}, methods={}}
Added field: l of type ClassType(Tree) to class Visitor
Added field: r of type ClassType(Tree) to class Visitor
Added method: visit of type INT to class Visitor
Added local variable: nti of type INT to method visit
Exception in thread "main" java.lang.NullPointerException
	at myVisitor.SymbolTableBuilderVisitor.visit(SymbolTableBuilderVisitor.java:93)
	at myVisitor.SymbolTableBuilderVisitor.visit(SymbolTableBuilderVisitor.java:19)
	at minijava.syntaxtree.MethodDeclaration.accept(MethodDeclaration.java:85)
	at minijava.visitor.GJVoidDepthFirst.visit(GJVoidDepthFirst.java:31)
	at minijava.syntaxtree.NodeListOptional.accept(NodeListOptional.java:46)
	at minijava.visitor.GJVoidDepthFirst.visit(GJVoidDepthFirst.java:150)
	at minijava.syntaxtree.ClassExtendsDeclaration.accept(ClassExtendsDeclaration.java:65)
	at minijava.syntaxtree.NodeChoice.accept(NodeChoice.java:35)
	at minijava.visitor.GJVoidDepthFirst.visit(GJVoidDepthFirst.java:113)
	at minijava.syntaxtree.TypeDeclaration.accept(TypeDeclaration.java:34)
	at minijava.visitor.GJVoidDepthFirst.visit(GJVoidDepthFirst.java:31)
	at minijava.syntaxtree.NodeListOptional.accept(NodeListOptional.java:46)
	at minijava.visitor.GJVoidDepthFirst.visit(GJVoidDepthFirst.java:63)
	at minijava.syntaxtree.Goal.accept(Goal.java:45)
	at Typecheck.main(Typecheck.java:15)
